### PR TITLE
drop messages instead of setting them to empty struct

### PR DESCRIPTION
**Pull request by karimra**
#36

### CODE REVIEW

File: formatters/event_drop/event_drop.go [hunk 1]
```diff
@@ -180,3 +185,15 @@ func (d *Drop) WithLogger(l *log.Logger) {
...
+	lastDropIndexes := len(dropIndexes) - 1
+	for i, dropIndex := range dropIndexes {
+		if i == lastDropIndexes {
+			es = es[:dropIndex]
+			continue
+		}
+		copy(es[dropIndex:], es[dropIndexes[i+1]:])
+	}
+	return es
+}
```
Comment (hellt) [inline formatters/event_drop/event_drop.go:199]: @karimra is the idea of this function to drop elements from `es` whos indexes match the values in `dropIndexes`?

e.g.
```go
dropIndexes := []int{0, 2, 3}
es := []string{"A", "B", "C", "D", "E", "F"}
// should return B E F ?
```

Comment (karimra) [reply formatters/event_drop/event_drop.go:199]: Yes but this doesn't seem to work properly when the first message is to be dropped, will rework it.

Comment (hellt) [reply formatters/event_drop/event_drop.go:199]: yep, I personally find this solution a very easy to read

https://goplay.tools/snippet/fcH6_GTddi1

it has a O(n) complexity, but it seems that the order of the events needs to be preserved, thus the re-slicing (shifting) is not considerably faster on the number of events

Comment (karimra) [reply formatters/event_drop/event_drop.go]: My point was to not have to allocate a new event slice to append the allowed messages.

**Review by hellt — APPROVED** [review]

**Pull request merged**